pages/api/spacehub/payment/index.ts

```diff
@@ -38,8 +42,24 @@ export default async function handler(
   switch (req.method) {
     case 'GET':
       try {
+        const session = await getServerSession(req, res, authOptions)
+        const user = await User.findOne({ email: session.user.email })
         const payments = await Payment.find({})
-        return res.status(200).json({ success: true, data: payments })
+          .sort({ date: -1 })
+          .limit(5)
+          .populate('payer')
+
+        let userPayments
+        if (user.role !== Roles.ADMIN) {
+          userPayments = payments.filter((p) => {
+            return p.payer.email === user.email
+          })
+        }
+
+        return res.status(200).json({
+          success: true,
+          data: user.role === Roles.ADMIN ? payments : userPayments,
```

Comment:
can be create a variable: isAdmin= user.role === Roles.ADMIN
user.role === Roles.ADMIN ==>
const isAdmin=user.role === Roles.ADMIN

user.role !== Roles.ADMIN==> !isAdmin